src/vectorpostprocessors/ResponseHistoryMean.C

```diff
@@ -0,0 +1,81 @@
+// MOOSE includes
+#include "ResponseHistoryMean.h"
+#include "PostprocessorInterface.h"
+#include "VectorPostprocessorInterface.h"
+#include "MastodonUtils.h"
+#include "ResponseHistoryBuilder.h"
+/* #include <iostream>
+#include <fstream>
+using namespace std;*/
+
+registerMooseObject("MastodonApp", ResponseHistoryMean);
+
+template <>
+InputParameters
+validParams<ResponseHistoryMean>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  params.addRequiredParam<VectorPostprocessorName>(
+      "vectorpostprocessor",
+      "Name of the ResponseHistoryBuilder vectorpostprocessor, for which "
+      "response spectra are calculated.");
+
+  // Make sure that csv files are created only at the final timestep
+  params.set<bool>("contains_complete_history") = true;
+  params.suppressParameter<bool>("contains_complete_history");
+
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_FINAL};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  params.addClassDescription("Calculate the mean acceleration time series given a set of nodes.");
+  return params;
+}
+
+ResponseHistoryMean::ResponseHistoryMean(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters),
+    _history_time(getVectorPostprocessorValue("vectorpostprocessor", "time"))
+
+{
+}
+
+void
+ResponseHistoryMean::initialSetup()
+{
+  const ResponseHistoryBuilder & history_vpp = getUserObjectByName<ResponseHistoryBuilder>(
+      getParam<VectorPostprocessorName>("vectorpostprocessor"));
+  std::vector<std::string> history_names =
+      history_vpp.getHistoryNames(); // names of the vectors in responsehistorybuilder
+  _history_acc.resize(history_names.size());
+
+  for (std::size_t i = 0; i < history_names.size(); i++)
+  {
+    // cout << history_names[i] << endl;
+    _history_acc[i] = history_vpp.getHistories()[i];
+  }
+
+  // Declaring a vector of pointers for storing the addresses of the time vector
+  // and the mean response history vector.
+
+  _mean_acc.push_back(&declareVector("_time"));
+  _mean_acc.push_back(&declareVector("_mean"));
+}
+
+void
+ResponseHistoryMean::initialize()
+{
+  for (VectorPostprocessorValue * ptr : _mean_acc)
+    ptr->clear();
+}
+
+void
+ResponseHistoryMean::execute()
+{
+
+  // Calling the "mean" (overloaded) function to compute the mean of response
+  // histories.
+
+  std::vector<std::vector<Real>> vec_mean_acc = MastodonUtils::mean(_history_acc, _history_time);
```

Comment:
```
_history_time = _builder.getVector("time");
_history_mean = MastodonUtils::mean(_builder.getHistories());
```

Reply:
This is what I am doing now, with a tiny difference. I want both the mean history and the time as outputs. So, I am creating a vector of pointers called _mean_acc where the first pointer stores the address of time and the second pointer stores the address of the mean history. _mean_acc is sent as output.

Reply:
Why do you want the time as an output, it is known and doesn't affect the calculation? Doing so will also create additional copies of the vector. If you were to actually want two outputs you should use a std::pair, but again you don't need it here.

Reply:
If I didn't have time as an output, in the excel output file, I only have a vector of the mean history. While the excel file of all the response histories (created by ResponseHistoryBuilder class) does have a time column, I thought it would be nice to have a time column in the mean history excel file as well. 

Is there a simple way to do this other than what I did?

Reply:
_history_time = _builder.getVector("time"); is giving me the following error. 

_**error: 'this' argument to member function 'getVector' has type 'const ResponseHistoryBuilder', but function is not marked const**_

After going through some posts on stackoverflow, it seems to me that the above requires modifying some function to const in the vectorpostprocessor class. Not wanting to do that, I implemented a workaround.

Reply:
This might be a design flaw in our interface, I will fix that in MOOSE. But, in the interim you can create get a reference to this VPP in the initialization list, and then to the assignment.

In the .h file:

const VectorPostprocessorData & _builder_time;

In the .C file initialization list

_builder_time(getVectorPostprocessorValue(...))

Then on the line that is causing you problems:

_history_time = _builder_time;

Reply:
Yes, this is exactly what I have done, but with different variable names.